From 2520f4528742decf78a8b375f5389b50977f5e4b Mon Sep 17 00:00:00 2001
From: Isaku Yamahata <yamahata@valinux.co.jp>
Date: Fri, 7 Dec 2012 12:37:02 +0900
Subject: [PATCH] linux/Makefile.main.in, acinclude: preparation for linux 3.7.0+

The version.h is moved from include/linux/version.h to
include/generated/uapi/linux/version.h.
So check both pathes.

Signed-off-by: Isaku Yamahata <yamahata@valinux.co.jp>
Signed-off-by: Jesse Gross <jesse@nicira.com>
---
 acinclude.m4                    |    3 ++-
 datapath/linux/Makefile.main.in |    5 ++++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/acinclude.m4 b/acinclude.m4
index d502b00..02dc952 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -144,7 +144,8 @@ AC_DEFUN([OVS_CHECK_LINUX], [
          AC_ERROR([Linux kernel in build tree $KBUILD (source tree $KSRC) is version $kversion, but version 2.6 or later is required])
        fi
     fi
-    if test ! -e "$KBUILD"/include/linux/version.h || \
+    if (test ! -e "$KBUILD"/include/linux/version.h && \
+        test ! -e "$KBUILD"/include/generated/uapi/linux/version.h)|| \
        (test ! -e "$KBUILD"/include/linux/autoconf.h && \
         test ! -e "$KBUILD"/include/generated/autoconf.h); then
 	AC_MSG_ERROR([Linux kernel source in $KBUILD is not configured])
diff --git a/datapath/linux/Makefile.main.in b/datapath/linux/Makefile.main.in
index 6b46c0c..9df3ad1 100644
--- a/datapath/linux/Makefile.main.in
+++ b/datapath/linux/Makefile.main.in
@@ -53,7 +53,10 @@ endif
 
 VERSION_FILE := $(KOBJ)/include/linux/version.h
 ifeq (,$(wildcard $(VERSION_FILE)))
-  $(error Linux kernel source not configured - missing version.h)
+  VERSION_FILE := $(KOBJ)/include/generated/uapi/linux/version.h
+  ifeq (,$(wildcard $(VERSION_FILE)))
+    $(error Linux kernel source not configured - missing version.h)
+  endif
 endif
 
 CONFIG_FILE := $(KSRC)/include/generated/autoconf.h
-- 
1.7.1

